__tests__/app.test.js

```diff
@@ -188,3 +188,77 @@ describe("GET/api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST/api/articles/:article_id/comments", () => {
+  it("POST returns 201 status with the a named object containing the posted comment ", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "I hate streaming eyes even more",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        const { postComment } = body;
+        expect(typeof body).toBe("object");
+        expect(postComment).toMatchObject({
+          comment_id: expect.any(Number),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+          author: "icellusedkars",
+          body: "I hate streaming eyes even more",
+          article_id: 1,
+        });
+      });
+  });
+
+  test("POST 201: should ignore unnecessary properties in the input", () => {
+    const input = {
+      username: "icellusedkars",
+      body: "I hate streaming eyes even more",
+      tagline: "my cool comment",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(input)
+      .expect(201)
+      .then(({ body }) => {
+        const { postComment } = body;
+        expect(postComment).not.toHaveProperty("tagline");
+      });
+  });
+
+    test("return a 404 status error message when new comment is passed a key with a value of the wrong data type", () => {
+      const newComment = {
+        username: "icellusedkars",
+        body: 10,
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+          const {msg} = body
+          expect(msg).toBe("bad request");
+        });
+    });
+
+    test("returns 400 error message when given an invalid id", () => {
```

Comment:
It would also have been nice to see a test for no username/body property, and a 404 for valid but non-existent article